resources-master/online-shop/src/app/product-list/product-list.component.ts

```diff
@@ -0,0 +1,31 @@
+import { Component, OnInit } from '@angular/core';
+import { ProductHeader } from '../_models/product-header.model';
+import { HttpClient } from '@angular/common/http';
+import { ProductService } from '../_services/product.service';
+
+@Component({
+  selector: 'app-product-list',
+  templateUrl: './product-list.component.html',
+  styleUrls: ['./product-list.component.css']
+})
+export class ProductListComponent implements OnInit{
+
+  productHeaders : ProductHeader[] = [
```

Comment:
The same discussion regarding the smart components.

Also, why do we have mock data in here?

Also, why is the HTTP client injected in this class?